phoenix-core/src/main/java/org/apache/phoenix/expression/LikeExpression.java

```diff
@@ -289,8 +288,11 @@ public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
             return true;
         }
 
-        String value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
-        boolean matched = pattern.matcher(value).matches();
+        String value = null;
+        if (logger.isDebugEnabled()) {
+            value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
```

Comment:
This is just for debugging I'm assuming, as we don't want to turn ptr into a String. Also, we should rename getStrExpression().